src/Athena/Controllers/api/RequirementController.cs

```diff
@@ -0,0 +1,136 @@
+using System.Net;
+
+namespace Athena.Controllers.api
+{
+    [Route("api/v1/[Controller]")]
+    public class RequirementController : Controller
+    {
+      private readonly IRequirementRepository requirements;
+      private readonly IProgramRepository programs;
+      private readonly ICourseRepository courses;
+
+        public RequirementController (IRequirementRepository requirementRepository, IProgramRepository programsRepository, ICourseRepository courseRepository)
+        {
+            requirements = requirementRepository ?? throw new ArgumentNullException(nameof(requirementRepository));
+            programs = programsRepository ?? throw new ArgumentNullException(nameof(programsRepository));
+            courses = courseRepository ?? throw new ArgumentNullException(nameof(courseRepository));
+        }
+
+        [HttpPost]
+        public async Task AddRequirement([FromBody] Requirement requirement) => await requirements.AddAsync(requirement);
+
+        [HttpGet("{id}")]
+        public async Task<Requirement> GetRequirement(Guid id) => await requirements.GetAsync(id);
+
+        [HttpPut("{id}")]
+        public async Task EditRequirement(Guid id, [FromBody] Requirement requirement)
+        {
+            if (id != requirement.Id)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to edit {id} but got a model for {requirement.Id}");
+            }
+            await requirements.EditAsync(requirement);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task DeleteRequirement(Guid id)
+        {
+            var requirement = await requirements.GetAsync(id);
+            if (requirement == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to delete {requirement} that does not exist");
+            }
+            await requirements.DeleteAsync(requirement);
+        }
+
+        [HttpPost("api/v1/course/{id}/requirement/satisfies/{reqId}")]
+        public async Task AddSatisfiedRequirementAsync(Course course, Requirement requirement)
+        {
+            if (requirement == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add satisfied requirements {requirement} to course {course} that doesn't exist");
+            }
+            await courses.AddSatisfiedRequirementAsync(course, requirement);
+        }
+
+        [HttpDelete("api/v1/course/{id}/requirement/satisfies/{reqId}")]
+        public async Task RemoveSatisfiedRequirementAsync(Course course, Requirement requirement)
+        {
+            if (requirement == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to remove satisfied requirements {requirement} to course {course} that doesn't exist");
+            }
+            await courses.RemoveSatisfiedRequirementAsync(course, requirement);
+        }
+
+        [HttpPost("api/v1/course/{id}/requirement/prereq/{reqId}")]
```

Comment:
Did you intend to implement this here or in a composite controller? If you intend to implement it here you need to use `/api/v1/student...` to override the controller-wide prefix.